src/core/mesh/qgsmeshdatasetgroupstore.cpp

```diff
@@ -315,6 +316,15 @@ QDomElement QgsMeshDatasetGroupStore::writeXml( QDomDocument &doc, const QgsRead
     ++it;
   }
 
+  for ( auto it = mGroupNameToGlobalIndex.constBegin(); it != mGroupNameToGlobalIndex.constEnd(); ++it )
+  {
+    QDomElement elemNameToIndex = doc.createElement( QStringLiteral( "name-to-global-index" ) );
+    elemNameToIndex.setAttribute( QStringLiteral( "name" ), it.key() );
+    elemNameToIndex.setAttribute( QStringLiteral( "global-index" ), it.value() );
```

Comment:
Wondering if we need to store global-index at all. I am thinking to store in XML only dataset group names, and use the dataset group names also in all other places in XML when you need to store something (like renderer settings). The actual index (int) could be then anything, important is that the group name will be associated with the correct renderer?

Reply:
I think ORG settings are stored this way?